app/src/main/java/org/simple/clinic/instantsearch/InstantSearchEffectHandler.kt

```diff
@@ -73,6 +74,20 @@ class InstantSearchEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun checkIfPatientAlreadyHasAnExistingNHID(): ObservableTransformer<CheckIfPatientAlreadyHasAnExistingNHID, InstantSearchEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .flatMap { patientId -> loadPatientProfileFromId(patientId) }
+          .map { patientProfile -> patientProfile.get().withoutDeletedBusinessIds() }
+          .map { if (it.hasNationalHealthID) PatientAlreadyHasAnExistingNHID else PatientDoesNotHaveAnExistingNHID(it.patientUuid) }
+    }
+  }
+
+  private fun loadPatientProfileFromId(it: CheckIfPatientAlreadyHasAnExistingNHID) =
+      patientRepository
+          .patientProfile(it.patientId)
+          .subscribeOn(schedulers.io())
+
```

Comment:
Hmm, this hasn't changed anything from earlier, we are still subscribing to change notifications here.

What I meant was to just make it a synchronous call without needing to subscribe to changes, like this:

```kotlin
return ObservableTransformer { effects ->
      effects
          .observeOn(schedulers.io())
          .map { patientRepository.patientProfileImmediate(it.patientId) }
          .map { patientProfile -> patientProfile.get().withoutDeletedBusinessIds() }
          .map { if (it.hasNationalHealthID) PatientAlreadyHasAnExistingNHID else PatientDoesNotHaveAnExistingNHID(it.patientUuid) }
    }
```

Reply:
oh yes, sorry. Just pushed the branch with everything.